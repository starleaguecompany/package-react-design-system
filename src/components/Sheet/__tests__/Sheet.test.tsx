import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Sheet } from '../index';

describe('Sheet', () => {
  test('should render correctly with no props', async () => {
    const { container } = render(<Sheet />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId('Sheet')).not.toBeInTheDocument();
  });

  test('should render correctly with visible prop', async () => {
    render(<Sheet visible />);

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toHaveClass('container', { exact: true });
    expect(screen.getByTestId('Sheet')).toBeVisible();
  });

  test('should render correctly with attributes', () => {
    render(<Sheet id="test-id" className="test-class" visible />);

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toHaveClass('test-class');
    expect(screen.getByTestId('Sheet')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with content', async () => {
    render(
      <Sheet visible>
        <Sheet.Content>Test content</Sheet.Content>
      </Sheet>
    );

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toHaveTextContent('Test content');
    expect(screen.getByText('Test content')).toBeVisible();
    expect(screen.getByText('Test content')).toHaveClass('content');
  });

  test('should render correctly with title prop', async () => {
    render(
      <Sheet visible>
        <Sheet.Header title="Test title" />
      </Sheet>
    );

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toContainElement(screen.getByText('Test title'));
    expect(screen.getByText('Test title')).toBeVisible();
    expect(screen.getByText('Test title')).toHaveClass('title');
  });

  test('should render correctly with subtitle prop', async () => {
    render(
      <Sheet visible>
        <Sheet.Header subtitle="Test subtitle" />
      </Sheet>
    );

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toContainElement(screen.getByText('Test subtitle'));
    expect(screen.getByText('Test subtitle')).toBeVisible();
    expect(screen.getByText('Test subtitle')).toHaveClass('subtitle');
  });

  test('should render correctly with closable prop', async () => {
    render(<Sheet visible closable />);

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toContainElement(screen.getByTestId('Cross'));
    expect(screen.getByTestId('Cross')).toBeVisible();
  });

  test('should render correctly with footer prop', async () => {
    render(
      <Sheet visible>
        <Sheet.Footer>Test footer</Sheet.Footer>
      </Sheet>
    );

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    expect(screen.getByTestId('Sheet')).toContainElement(screen.getByText('Test footer'));
    expect(screen.getByText('Test footer')).toBeVisible();
    expect(screen.getByText('Test footer')).toHaveClass('footer');
  });

  test('should execute onClose callback on close button click', async () => {
    const onCloseCallback = jest.fn();
    render(<Sheet visible closable onClose={onCloseCallback} />);

    expect(screen.getByTestId('Sheet')).toMatchSnapshot();
    userEvent.click(screen.getByTestId('Cross'));
    expect(onCloseCallback).toBeCalledTimes(1);
  });
});
