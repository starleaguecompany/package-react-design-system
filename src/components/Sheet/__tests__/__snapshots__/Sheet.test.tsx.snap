// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sheet should execute onClose callback on close button click 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
>
  <span
    class="close container size-20 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Sheet should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="Sheet"
  draggable="false"
  id="test-id"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
/>
`;

exports[`Sheet should render correctly with closable prop 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
>
  <span
    class="close container size-20 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</div>
`;

exports[`Sheet should render correctly with content 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
>
  <div
    class="content"
  >
    Test content
  </div>
</div>
`;

exports[`Sheet should render correctly with footer prop 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
>
  <div
    class="footer container direction-horizontal justify-stretch"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    Test footer
  </div>
</div>
`;

exports[`Sheet should render correctly with no props 1`] = `<div />`;

exports[`Sheet should render correctly with subtitle prop 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
>
  <div
    class="header"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    >
      <div
        class="subtitle"
      >
        Test subtitle
      </div>
    </div>
  </div>
</div>
`;

exports[`Sheet should render correctly with title prop 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
>
  <div
    class="header"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    >
      <div
        class="title"
      >
        Test title
      </div>
    </div>
  </div>
</div>
`;

exports[`Sheet should render correctly with visible prop 1`] = `
<div
  class="container"
  data-qa="Sheet"
  draggable="false"
  style="transform: translateY(1000px) translateZ(0); user-select: none;"
/>
`;
